# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1455726942 18000
#      Wed Feb 17 11:35:42 2016 -0500
# Node ID 952854b61ac93de5ea844fec8270d2bf727dd4db
# Parent  1a21a0f6364e48e2fdf834f486b7dd917e3326dc
don't include config.h in public header files

* graphics.in.h: Don't include config.h header.
* MSparse.h: Don't include config.h or functional headers.

diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -20,20 +20,16 @@ along with Octave; see the file COPYING.
 
 */
 
 #if ! defined (octave_graphics_h)
 #define octave_graphics_h 1
 
 #include "octave-config.h"
 
-#ifdef HAVE_CONFIG_H
-#  include <config.h>
-#endif
-
 #include <cctype>
 
 #include <algorithm>
 #include <list>
 #include <map>
 #include <set>
 #include <sstream>
 #include <string>
diff --git a/liboctave/array/MSparse.h b/liboctave/array/MSparse.h
--- a/liboctave/array/MSparse.h
+++ b/liboctave/array/MSparse.h
@@ -21,22 +21,16 @@ along with Octave; see the file COPYING.
 
 */
 
 #if ! defined (octave_MSparse_h)
 #define octave_MSparse_h 1
 
 #include "octave-config.h"
 
-#ifdef HAVE_CONFIG_H
-#  include <config.h>
-#endif
-
-#include <functional>
-
 #include "quit.h"
 #include "lo-error.h"
 #include "Sparse.h"
 #include "MArray.h"
 #include "Array-util.h"
 
 
 // Two dimensional sparse array with math ops.
